### PR TITLE
Update evalArgs to match the new planned API

### PR DESCRIPTION
**Pull request by Pacman99**
This simply encodes the public facing portion of the api #232. @blaggacao and I ended up changing the API after some discussion, so this is the most updated version of the planned API.

### CODE REVIEW

**Comment by Pacman99** [comment]
This has now been tested in nix repl and it evaluates all arguments correctly.

**Comment by blaggacao** [comment]
So go ahead and log it into `api-next`, I'd say!

File: lib/mkFlake/evalArgs.nix [hunk 1]
```diff
@@ -1,150 +1,229 @@
...
+            '';
+          };
+          modules = mkOption {
+            type = pathTo moduleType;
+            default = [ ];
+            description = ''
+              The configuration for this config
+            '';
+          };
+          externalmodules = mkOption {
```
Comment (blaggacao) [inline lib/mkFlake/evalArgs.nix:111]: `externalmodules` -> `externalModules`

**Review by blaggacao — APPROVED** [review]
A typo, that's all. This reads already super clean as it is.

Thank you for this great piece of work!

**Comment by Pacman99** [comment]
I'll commit some more fixups soon. And I'll send a PR to add the generated docs.

**Comment by Pacman99** [comment]
fixups in https://github.com/divnix/devos/commit/2b70cd3ae607d8c5d05b4266530be6011e94371b

**Pull request merged**